syntax = "proto3";

option go_package = "pb/game";

message MissionPBComp
{
	enum eMissionStatus
	{
	    E_MISSION_NORMAL = 0;
	    E_MISSION_ACHIEVEMENT = 1;
	    E_MISSION_COMPLETE = 2;
	    E_MISSION_TIME_OUT = 3;
	    E_MISSION_FAILD = 4;
	};
	uint32 id = 1;
	uint32 status = 2;
	repeated uint32 progress = 3;
}

message MissionListPBComp
{
	enum eMissionType
	{
		kPlayerMission = 0;//普通任务
		kPlayerAchievment = 1;//成就
		kPlayerDailyActivity = 2;//每日活动
		kPlayerMissionSize = 3;//最大值
	}
	map<uint32, MissionPBComp> missions = 1;
	map<uint32, bool> complete_missions = 2;
	map<uint32, uint64> mission_begin_time = 3;
	uint32 type = 4;
}

message RewardListPBComp
{
	map<uint32, bool> can_reward_mission_id = 2;
}
